cmake/schemes/url_sha1_boost_library.cmake.in

```diff
@@ -243,6 +244,35 @@ if(ANDROID)
   file(
       APPEND ${boost_user_jam} "  : \"${boost_compiler}\" ${CMAKE_CXX_FLAGS}\n"
   )
+elseif(WIN32)
+  # Even though user.config.jam provides the path to archiver 'b2'
+  # tries to find it based on the path to the compiler. In this 
+  # process if the path contains any spaces or any other invalid 
+  # characters that windows use for its long names, it derives a 
+  # path which is incorrect. To solve this we convert the path to 
+  # 8.3 FileName format or windows short filename format.
+  set(conversion_cmd
+	  "cmd" 
+	  "/c" 
+	  "for" 
+	  "%A" 
+	  "in" 
+	  ("${boost_compiler}") 
+	  "do" 
+	  "@echo"
+	  "%~sA"
+	)
```

Comment:
It would nice to have a link about what this command do. I've found this SO answer: http://stackoverflow.com/a/10227715/2288008